__tests__/integration/helpers/browserExecutors.js

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// usage: const cacheKeys = await browser.executeAsync(getCacheKeys);
+exports.getCacheKeys = function getCacheKeys(done) {
+  // eslint-disable-next-line prefer-arrow-callback
+  caches.keys().then(function filterKeys(cacheKeys) {
```

Comment:
is `caches` available through `browser.executeAsync` context ?

Reply:
Yes! The `caches` API (CacheStorage) is available both in the browser and service worker. With `browser.executeAsync`, the block of code is serialized and ran directly in the browser context, so we can access `caches` from the callback.